tools/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/monitoring"
+)
+
+// constant parts of the file
+const (
+	title      = "# Cluster Network Addons Operator metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by the Cluster Network Addons Operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool `tools/metricsdocs` and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## Cluster Network Addons Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run `make generate-doc` to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change `tools/metricsdocs` with your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+func main() {
+	metricsList := metricsOptsToMetricList(monitoring.MetricsOptsList)
+	sort.Sort(metricsList)
+	writeToFile(metricsList)
+}
+
+func writeToFile(metricsList metricList) {
+	fmt.Print(opening)
+	metricsList.writeOut()
+	fmt.Print(footer)
+}
+
+type metric struct {
+	name        string
+	description string
+}
+
+func metricsOptsToMetricList(mdl map[string]monitoring.MetricsOpts) metricList {
+	res := make([]metric, 0)
+	for _, element := range mdl {
+		res = append(res, metricDescriptionToMetric(element))
+	}
+
+	return res
+}
+
+func metricDescriptionToMetric(rrd monitoring.MetricsOpts) metric {
+	return metric{
+		name:        rrd.Name,
+		description: rrd.Help,
+	}
+}
+
+func (m metric) writeOut() {
+	fmt.Println("###", m.name)
+	fmt.Println(m.description)
+}
+
+type metricList []metric
+
+// Len implements sort.Interface.Len
+func (m metricList) Len() int {
+	return len(m)
+}
+
+// Less implements sort.Interface.Less
+func (m metricList) Less(i, j int) bool {
+	return m[i].name < m[j].name
+}
+
+// Swap implements sort.Interface.Swap
+func (m metricList) Swap(i, j int) {
+	m[i], m[j] = m[j], m[i]
+}
```

Comment:
can we use sort.Strings(strs) instead ?

Reply:
We are not sorting strings but a structure. It would surely be possible to find a way to use `sort.Strings` instead but I think the code would become more complicated.